hugegraph-core/src/main/java/org/apache/hugegraph/task/EphemeralJobQueue.java

```diff
@@ -0,0 +1,239 @@
+            this.queueWeakReference = new WeakReference<>(queue);
+        }
+
+        @Override
+        public String type() {
+            return BATCH_EPHEMERAL_JOB;
+        }
+
+        @Override
+        public Object execute() throws Exception {
+            boolean stop = false;
+            Object result = null;
+            int consumeCount = 0;
+            InterruptedException interruptedException = null;
+            EphemeralJobQueue queue;
+            List<EphemeralJob<?>> batchJobs = new ArrayList<>();
+            while (!stop) {
+                if (interruptedException == null && Thread.currentThread().isInterrupted()) {
+                    interruptedException = new InterruptedException();
+                }
+
+                queue = this.queueWeakReference.get();
+                if (queue == null) {
+                    stop = true;
+                    continue;
+                }
+
+                if (queue.isEmpty() || consumeCount > MAX_CONSUME_COUNT ||
+                    interruptedException != null) {
+                    queue.consumeComplete();
+                    stop = true;
+                    if (!queue.isEmpty()) {
+                        queue.reScheduleIfNeeded();
+                    }
+                    continue;
+                }
+
+                try {
+                    while (!queue.isEmpty() && batchJobs.size() < PAGE_SIZE) {
+                        EphemeralJob<?> job = queue.poll();
+                        if (job == null) {
+                            continue;
+                        }
+                        batchJobs.add(job);
+                    }
+
+                    if (batchJobs.isEmpty()) {
+                        continue;
+                    }
+
+                    consumeCount += batchJobs.size();
+                    result = this.executeBatchJob(batchJobs, result);
+
+                } catch (InterruptedException e) {
+                    interruptedException = e;
+                } finally {
+                    batchJobs.clear();
+                }
+            }
+
+            if (interruptedException != null) {
+                Thread.currentThread().interrupt();
+                throw interruptedException;
+            }
+
+            return result;
+        }
+
+        private Object executeBatchJob(List<EphemeralJob<?>> jobs, Object prevResult) throws Exception {
+            GraphIndexTransaction graphTx = this.params().systemTransaction().indexTransaction();
+            GraphIndexTransaction systemTx = this.params().graphTransaction().indexTransaction();
+            Object result = prevResult;
+            for (EphemeralJob<?> job : jobs) {
+                this.initJob(job);
+                Object obj = job.call();
+                if (job instanceof Reduce) {
+                    result = ((Reduce) job).reduce(result, obj);
+                }
+            }
+
+            graphTx.commit();
```

Comment:
can we call this.params().graphTransaction().commit() instead here?

Reply:
no, we can't, because RemoveLeftIndexJob use GraphIndexTransaction, it is only update index, so if use GraphTransaction, the commit will only return because update data is empty(but index data no empy)